great_expectations/experimental/datasources/interfaces.py

```diff
@@ -362,6 +361,16 @@ def _execution_engine_type(self) -> Type[ExecutionEngine]:
         """Returns the execution engine to be used"""
         return self.execution_engine_override or self.execution_engine_type
 
+    @functools.lru_cache(maxsize=1)
+    def _get_execution_engine(self, **engine_kwargs) -> ExecutionEngine:
```

Comment:
You may want to create a `clear_exec_engine_cache` fixture for use in testing.
```python
@pytest.fixture
def clear_exec_engine_cache():
     # maybe you want to inspect the cache before cleaning it??
    print(_get_execution_engine.cache_info())
    _get_execution_engine.clear_cache()
   return
```

https://docs.python.org/3/library/functools.html#functools.lru_cache